tools/dds/realdds-py/python.cpp

```diff
@@ -348,32 +359,17 @@ PYBIND11_MODULE(NAME, m) {
         .def( "guid", []( dds_device const & self ) { return to_string( self.guid() ); } )
         .def( "is_running", &dds_device::is_running )
         .def( "run", &dds_device::run )
-        .def( "num_of_sensors", &dds_device::num_of_sensors )
+        .def( "num_of_streams", &dds_device::number_of_streams )
         .def( FN_FWD( dds_device,
-            foreach_sensor,
-            ( size_t, std::string const & ),
-            ( size_t sensor_index, std::string const & name ),
-            callback( sensor_index, name ); ) )
-        .def( "foreach_video_profile",
-              []( dds_device const & self,
-                  size_t sensor_index,
-                  std::function< void( rs2_video_stream const & profile, bool is_default ) > callback ) {
-                  self.foreach_video_profile(
-                      sensor_index,
-                      [callback]( rs2_video_stream const & profile, bool is_default ) {
-                          FN_FWD_CALL( dds_device, foreach_video_profile, callback( profile, is_default ); );
-                      } );
-              }, py::call_guard< py::gil_scoped_release >() )
-        .def( "foreach_motion_profile",
-              []( dds_device const & self,
-                  size_t sensor_index,
-                  std::function< void( rs2_motion_stream const & profile, bool is_default ) > callback ) {
-                  self.foreach_motion_profile(
-                      sensor_index,
-                      [callback]( rs2_motion_stream const & profile, bool is_default ) {
-                          FN_FWD_CALL( dds_device, foreach_motion_profile, callback( profile, is_default ); );
-                      } );
-              }, py::call_guard< py::gil_scoped_release >() )
+                      foreach_stream,
+                      ( std::shared_ptr< dds_stream > ),
+                      ( std::shared_ptr< dds_stream > stream ),
+                      callback( stream ); ) )
+        //.def( FN_FWD( dds_device,
```

Comment:
Can you explain the changes here?
Why is this code commented out?

Reply:
Eran is already making changes that will make mine obsolete so I just changed what was needed for my compilation and commented out stuff that was not. Eran will handle this in his upcoming PR